src/hotspot/share/opto/callnode.cpp

```diff
@@ -1303,6 +1303,40 @@ void CallLeafVectorNode::calling_convention( BasicType* sig_bt, VMRegPair *parm_
 
 
 //=============================================================================
+bool CallLeafPureNode::is_unused() const {
```

Comment:
Can you add a quick comment why this check implies that the node is not used, i.e. what that means?

Reply:
I suppose I was not aware that `TypeFunc::Parms` stands for result projection.... the name does not make it immediately apparent.

Reply:
I think I was really just confused about the `Parms`. I thought that means parameters .. and not results :rofl: